### PR TITLE
Fix accept loop exception handling

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

**Comment by edsko** [comment]
@coot Cancelled bors because the tests seem to loop with this PR, causing Hydra to time out and bors to stall all other PRs. Sorry.

File: ouroboros-network/src/Ouroboros/Network/Socket.hs [hunk 1]
```diff
@@ -485,5 +485,4 @@ withServerNode muxTracer handshakeTracer errorPolicyTracer tbl stVar addr encode
 
       acceptException :: Socket.SockAddr -> SomeException -> IO ()
       acceptException a e = do
-        traceWith (WithAddr a `contramap` errorPolicyTracer) $ ErrorPolicyAccept e
-        throwIO e
+        traceWith (WithAddr a `contramap` errorPolicyTracer) $ ErrorPolicyAcceptException e
```
Comment (avieth) [inline ouroboros-network/src/Ouroboros/Network/Socket.hs]: Must rethrow async exceptions, or at the very least, `AsyncCancelled` and `ThreadKilled` exceptions.

Comment (avieth) [reply ouroboros-network/src/Ouroboros/Network/Socket.hs]: This is the exception handler around `Network.Socket.accept` right? I'd recommend only catching `IOExceptions` here.

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Socket.hs]: Yes, we should only catch `IOException`s here.

**Pull request merged**